### DISCUSSION TITLE
Implement Driving Directions, Walk and Cycling using react-map-gl

### DISCUSSION THREAD

**Discussion by tuscanlabs** [uber/react-map-gl#1379] [Q&A]
Hi,

I really like using react-map-gl.  I wanted to implement driving directions.  Is there a easy way to do that.  Currently my code is:
     <ReactMapGL  {...viewport} 
      mapboxApiAccessToken={props.token}
      mapStyle='mapbox://styles/mapbox/streets-v11'
      onViewportChange={(viewport) => {setViewport(viewport);}} 
      >
      <Marker key='one' longitude={-75.159257} latitude={39.953253}
      > 
      <Button
              className="marker-btn"
              onClick={e => {
                e.preventDefault();
                setSelectedPark(park);
              }}
            > 
            <img src="/images/maps.svg" alt="Skate Park Icon" /> 
            </Button>
      </Marker>
      <NavigationControl style={navControlStyle} />
      <GeolocateControl
        style={geolocateControlStyle}
        positionOptions={{enableHighAccuracy: true}}
        trackUserLocation={true}
        auto
      />
      
      </ReactMapGL>

**Comment by Pessimistress** [comment]
You can use [getMap](http://visgl.github.io/react-map-gl/docs/api-reference/static-map#getmap) to get the Map instance and then follow [the official example](https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-directions/).
